app/notification/application_submission/process_contents.py

```diff
@@ -0,0 +1,94 @@
+from dataclasses import dataclass
+from datetime import datetime
+from io import StringIO
+
+from app.notification.models.notification import NotificationData
+
+
+@dataclass
+class ApplicationData:
+    """Class process the application data to map with
+    APPLICATION_RECORD_OF_SUBMISSION template  from
+    govuk-notify service.
+
+    Returns:
+        Mapped data object from application contents with
+        govuk-notify service template & creates txt file
+        for applicant to download.
+    """
+
+    contact_info: str
+    questions: dict
+    submission_date: str
+    fund_name: dict
+    fund_round: dict
+    application_id: dict
+
+    @property
+    def format_submission_date(self):
+        return datetime.strptime(
+            self.submission_date, "%Y-%m-%d %H:%M:%S"
+        ).strftime("%Y-%m-%d")
+
+    @staticmethod
+    def from_json(data):
+        """Function calls ApplicationData class to map
+        the application contents.
+
+        Args:
+            data: Takes application data from NotificationData
+            class.
+
+        Returns:
+            ApplicationData object with application contents.
+        """
+        json_data = NotificationData.notification_data(data)
+        application = json_data.content["application"]
+        return ApplicationData(
+            contact_info=json_data.contact_info,
+            questions=ApplicationData.create_memory_object(json_data),
+            submission_date=application.get("date_submitted"),
+            fund_name=application.get("project_name"),
+            fund_round=application.get("round_id"),
+            application_id=application.get("id"),
+        )
+
+    @staticmethod
+    def get_sections(data) -> list:
+        sections = data.content["application"]["sections"]
+        return sections
+
+    @staticmethod
+    def get_section_names(data) -> list:
+        section_names = []
+        sections = ApplicationData.get_sections(data)
```

Comment:
This should be a normal method of the class perhaps, so you would have access to self, so you could do self.get_sections() etc rather than passing data and using ApplicationData..

Reply:
Thanks, Harry, I am going to leave it static method for now, will do it on the next sprint